feature(examples): added new example using OOP pattern for NPS API (#71)
fivetran/fivetran_connector_sdk: examples/quickstart_examples/oop_example/Readme.md

# National Parks Example

This example implements a **connector** to fetch, process, and store data from the [National Parks Service API (NPS API)](https://www.nps.gov/subjects/developer/index.htm). The process is built with an **Object-Oriented Programming (OOP)** approach, ensuring modular, maintainable, and reusable code. 

---

## Overview

> This example retrieves data from the **NPS API**, processes it, and stores it in a structured format across four tables:  

Reviewer: ```suggestion
This example retrieves data from the NPS API, processes it, and stores it in a structured format across four tables:  
```

> **Parks**, **Alerts**, **Articles**, and **People**. Each table is encapsulated within its own Python file, ensuring single responsibility and clean organization.

Reviewer: ```suggestion
 - Parks
- Alerts
- Articles
- People

Each table is encapsulated within its own Python file, ensuring single responsibility and clean organization.
```

The project follows a modular architecture:
- Each table has its own dedicated class for handling schema definitions and data transformations.

> - A centralized **NPS Client** handles API interactions, abstracting away the complexities of HTTP requests.

Reviewer: ```suggestion
- A centralized NPS Client handles API interactions, abstracting away the complexities of HTTP requests.
```

---

## Features

- **Modular Design:** Each table and the API client are encapsulated in separate files for clarity and reusability.
- **Scalable:** Easily extend it to accommodate additional tables or API endpoints.

> - **Customizable:** Modify transformations or table structures without affecting unrelated components.

Reviewer: ```suggestion
- Customizability - you can modify transformations or table structures without affecting unrelated components
```

> - **Reliable:** Includes error handling for API interactions and data processing.

Reviewer: ```suggestion
- Reliable - includes error handling for API interactions and data processing
```

---

## Project Structure

```plaintext
├── parks.py         # Handles the Parks table
├── alerts.py        # Handles the Alerts table
├── articles.py      # Handles the Articles table
├── people.py        # Handles the People table
├── nps_client.py    # Handles API initialization and data fetching
├── requirements.txt # Lists dependencies
├── README.md        # Project documentation
└── connector.py     # Main file to run the connector
```

## Run the Connector

```bash

> Run fivetran debug 

Reviewer: @orizwanft Is Run really a part of the command? I don't think so.

```

## Output

### Parks

> Contains detailed information about national parks

Reviewer: ```suggestion
This table contains detailed information about national parks
```

> ![PARKS](images/Parks.png "Parks Table in DB")

Reviewer: @orizwanft "Parks Table in DB" - what is this string for? It's not displayed in the GitHub preview. This relates to all screenshots in the file.

Reviewer: It's all good now, the screenshots are displayed.

#### Interact with Parks table

> 

Reviewer: ```suggestion

Run the following query:

```

```sql

select * from parks

```

### Articles

> Stores educational and informational articles about national parks

Reviewer: ```suggestion
This tables stores educational and informational articles about national parks.
```

![Articles](images/Articles.png "Articles Table in DB")

#### Interact with Articles table

> 

Reviewer: ```suggestion

Run the following query:

```

```sql

select * from articles

```

### Alerts

> Captures active alerts for parks

Reviewer: ```suggestion
This table captures active alerts for parks.
```

![Alerts](images/Alerts.png "Alerts Table in DB")

#### Interact with Alerts table

> 

Reviewer: ```suggestion

Run the following query:

```

```sql

select * from alerts

```
### People

Lists key figures associated with the parks or their history

![PEOPLE](images/People.png "People Table in DB")

#### Interact with People table

> 

Reviewer: ```suggestion

Run the following query:

```

- Reliability - includes error handling for API interactions and data processing

---

## Project Structure

```plaintext
├── parks.py         # Handles the Parks table
├── alerts.py        # Handles the Alerts table
├── articles.py      # Handles the Articles table
├── people.py        # Handles the People table
├── nps_client.py    # Handles API initialization and data fetching
├── requirements.txt # Lists dependencies
├── README.md        # Project documentation
└── connector.py     # Main file to run the connector
```

## Run the Connector

```bash
Run fivetran debug 
```

## Output

### Parks

This table contains detailed information about national parks

![PARKS](images/Parks.png "Parks Table in DB")

#### Interact with Parks table

Run the following query:

```sql

select * from parks

```

### Articles

This tables stores educational and informational articles about national parks.

![Articles](images/Articles.png "Articles Table in DB")

#### Interact with Articles table

Run the following query:

```sql

select * from articles

```

### Alerts

This table captures active alerts for parks.

![Alerts](images/Alerts.png "Alerts Table in DB")

> #### Interact with Alerts table

Reviewer: ```suggestion
#### Query the Alerts table
```

Run the following query:

```sql

select * from alerts

```
### People

This table lists key figures associated with the parks or their history.

![PEOPLE](images/People.png "People Table in DB")

> #### Interact with People table

Reviewer: ```suggestion
#### Query the People table
```

# National Parks Example

> This example implements a connector to fetch, process, and store data from the [National Parks Service API (NPS API)](https://www.nps.gov/subjects/developer/index.htm). The process is built with the Object-Oriented Programming (OOP) approach, ensuring modular, maintainable, and reusable code. 

Reviewer 2: ```suggestion
This example implements a connector to fetch, process, and store data from the [National Parks Service API (NPS API)](https://www.nps.gov/subjects/developer/index.htm). The process is built based on the Object-Oriented Programming (OOP) approach, ensuring modular, maintainable, and reusable code. 
```

---

## Overview

This example retrieves data from the NPS API, processes it, and stores it in a structured format across the following tables:  
- Parks
- Alerts
- Articles
- People

Each table is encapsulated within its own Python file, ensuring single responsibility and clean organization.

The project follows a modular architecture:
- Each table has its own dedicated class for handling schema definitions and data transformations.
- A centralized NPS Client handles API interactions, abstracting away the complexities of HTTP requests.

---

## Features

> - Modular Design - each table and the API client are encapsulated in separate files for clarity and reusability

Reviewer 2: ```suggestion
- Modular design - each table and the API client are encapsulated in separate files for clarity and reusability
```

- Scalability - you can easily extend it to accommodate additional tables or API endpoints
- Customizability - you can modify transformations or table structures without affecting unrelated components
- Reliability - includes error handling for API interactions and data processing

---

## Project Structure

```plaintext
├── parks.py         # Handles the Parks table
├── alerts.py        # Handles the Alerts table
├── articles.py      # Handles the Articles table
├── people.py        # Handles the People table
├── nps_client.py    # Handles API initialization and data fetching
├── requirements.txt # Lists dependencies
├── README.md        # Project documentation
└── connector.py     # Main file to run the connector
```

> ## Run the Connector

Reviewer 2: ```suggestion
## Run the connector
```
